src/clj/schema/experimental/generators.clj

```diff
@@ -26,56 +31,156 @@
     (fn [d] (generators/make-gen (fn [r s] (generators/call-gen @d r s))))
     (fn [] (subschema-generator schema params))))
 
-(defprotocol LeafGeneratable
-  (leaf-generator [s] "return a generator for s"))
+(defprotocol CompositeGenerator
+  (composite-generator [s params]))
 
-(extend-type nil
-  LeafGeneratable
-  (leaf-generator [x]
-    (cond
-     (= x s/Int) generators/int
-     (= x s/Str) generators/string-ascii
-     (= x s/Keyword) generators/keyword
-     :else (generators/return x))))
+(extend-protocol CompositeGenerator
+  schema.spec.variant.VariantSpec
+  (composite-generator [s params]
+    (generators/such-that
+     #(not ((:pre s) %))
```

Comment:
:sheep: (.-pre ^schema.spec.leaf.VariantSpec this) is safer. Same below.

also, maybe `(complement pre)` instead of `#(not (pre %))`